node_monitor/node_monitor.py

```diff
@@ -141,19 +146,23 @@ def broadcast_status_report(self) -> None:
                             if k in subscribers.keys()}
         # - - - - - - - - - - - - - - - - -
         for node_provider_id, nodes in reportable_nodes.items():
+            logging.info(f"Broadcasting status report {node_provider_id}...")
             preferences = subscribers[node_provider_id]
             subject, message = messages.nodes_status_message(nodes, node_labels)
             # - - - - - - - - - - - - - - - - -
             if preferences['notify_email'] == True:
                 recipients = email_recipients[node_provider_id]
+                logging.info(f"Sending status report email to {recipients}...")
                 self.email_bot.send_emails(recipients, subject, message)
             if preferences['notify_slack'] == True:
                 if self.slack_bot is not None:
                     channel_name = channels[node_provider_id]['slack_channel_name']
+                    logging.info(f"Sending status report slack message to {channel_name}...")
                     self.slack_bot.send_message(channel_name, message)
             if preferences['notify_telegram_chat'] == True: 
                 if self.telegram_bot is not None:
                     chat_id = channels[node_provider_id]['telegram_chat_id']
+                    logging.info(f"Sending status report telegram message to {chat_id}...")
                     self.telegram_bot.send_message(chat_id, message)
```

Comment:
Should we add logging in the method declaration as well. E.g. `slack_bot.send_message()` returns an error. So should we add logging in `bot_slack.py` as well? Same for `bot_telegram.py`?

Reply:
Hmm, in what way?

We could do something here like
```py
if self.telegram_bot is not None:
    chat_id = channels[node_provider_id]['telegram_chat_id']
    logging.info(f"Sending status report telegram message to {chat_id}...")
    err = self.telegram_bot.send_message(chat_id, message)
    if err is not None:
        logging.error(f"Failed to send status report telegram message to {chat_id}...")
```

Reply:
I was thinking more like:
```py
def send_message(
            self, slack_channel_name: str,
            message: str) -> None | SlackApiError:
        try:
            self.client.chat_postMessage(
                channel=slack_channel_name,
                text=message)
        except SlackApiError as e:
            # You will get a SlackApiError if "ok" is False
            # If ok is False, e.response["error"] contains a
            # str like 'invalid_auth', 'channel_not_found'
            assert e.response["ok"] is False
            assert e.response["error"]
            logging.error(f"Failed to send status report telegram message to {chat_id}...")
            return e
        return None
```
       
in `bot_slack.py`. And then the same in `bot_telegram.py`.  This way we can avoid nested `if` statements